app/controllers/simulations_controller.rb

```diff
@@ -205,6 +215,9 @@ def mark_as_complete
           else
             @experiment.progress_bar_update(@simulation_run.index, 'done')
           end
+
+          algorithm_runner = WorkersScaling::AlgorithmRunner.get(@experiment.id)
+          Thread.new { algorithm_runner.execute_and_schedule } if algorithm_runner
```

Comment:
To też można obłożyć logowaniem Exception lub StandardError, bo wywoływane jest w wątki i możemy nie zobaczyć ew. błędu.